### DISCUSSION TITLE
Use inside config

### DISCUSSION THREAD

**Discussion by hamzaelmaghari** [spatie/laravel-settings#144] [Q&A] [answered]
What should be done to use these settings inside config files such as ````config/app.php```` etc..

**Comment by rubenvanassche** [comment, marked as answer]
You overwrite the config by loading settings classes within the boot method of your AppServiceProvider for example.

**Reply by hamzaelmaghari** [reply]
Thanks, I did that. but, the question is:
Will that affect the performance? if having for example more than 100 settings?

Thanks :)

**Reply by rubenvanassche** [reply]
When caching settings on production, performance probably isn't going to be an issue.

**Reply by aecioneto** [reply]
Hi, there.
I have this requirement as well!
Tried many ways here without success. For sure, I have no idea on how to have Settings set before any config.

A code example can be provided, please? ;)

**Reply by hamzaelmaghari** [reply]
> Hi, there.
> I have this requirement as well!
> Tried many ways here without success. For sure, I have no idea on how to have Settings set before any config.
> 
> A code example can be provided, please? ;)

Basically you can add the following code in the `boot` method of your `AppServiceProvider`:

````
$setting = app(ExampleSettingClass::class)->toArray();

config()->set([
'app.name' => $setting->example,
// Repeat as many settings you have
]);
````

Apologies for any mistake cause I've writen that code using my phone 🤳😬

**Reply by aecioneto** [reply]
It didn' t work.
I believe there no way.
My requirement is to have some settings into other config/xxx.php files.
The best I could get is this result when running *php artisan*:
 Class "MySettings" does not exist 
 
All other tries, breaks artisan completely:
In SettingsRepositoryFactory.php line 14:
array_key_exists(): Argument #2 ($array) must be of type array, null given  

Thanks, anyway!!

**Reply by aecioneto** [reply]
Hello once more!
I finally made it.
There are some gotchas, as follow:
0. Create a new Service Provider, for instance:

```
class ConfigProvider extends ServiceProvider
{
    /**
     * Register services.
     */
    public function register(): void
    {
        //
    }

    /**
     * Bootstrap services.
     */
    public function boot(): void
    {
        $this->loadCustomerSettings();
    }

    public function loadCustomerSettings()
    {
        $theConfig = null;

        try
        {
            $theConfig = app(CustomerSettings::class);

            //just do anything for checking
            $dummy = $theConfig->coName;
        }
        catch (\Exception $e)
        {
            Log::error($e);

            return;
        }

        config()->set([
            'customer.name'    => "$theConfig->coName",
            'customer.address'    => "$theConfig->coAddr",
        ]);
    }
```

1. There must be a try/catch as shown above:
If migration is not done or there is no **coName** in the customer group inside settings table yet, it does not break artisan commands.

2. When no Exception is raised, config() is called to override settings you wish.

Best regards.

**Comment by shruko** [comment]
Set config('boilerplate.page_slug') in AppServiceProvider

Route::get(config('boilerplate.page_slug') . '/{page_slug}', [PageController::class, 'show'])
->name('page.show');

However, when I clear the cache, it defaults back to the value specified in the configuration file.

any solution?
